include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -461,39 +330,200 @@ class union_composition
 protected:
     //!\privatesection
 
-    //!\brief Converts an object of one of the given alphabets into the internal representation
+    /*!\brief Returns the max over all alphabet_size_v<alphabets_t>'s.
+     * \tparam ...alphabets_t The types must satisfy seqan3::alphabet_concept.
+     *
+     * ```cpp
+     * constexpr auto max = alphabet_max_size_v<dna4, gap, dna5>;
+     * assert(max == 5);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto alphabet_max_size_v =
+        std::max({static_cast<size_t>(0), static_cast<size_t>(alphabet_size_v<alphabets_t>)...});
+
+    /*!\brief Returns an array which contains the prefix sum over all alphabet_types::value_size's.
+     * \tparam ...alphabet_types The types must satisfy seqan3::alphabet_concept.
+     *
+     * ```cpp
+     * constexpr auto prefix_sum = alphabet_prefix_sum_sizes<dna4, gap, dna5>();
+     * assert(prefix_sum.size() == 4);
+     * assert(prefix_sum[0] == 0);
+     * assert(prefix_sum[1] == 4);
+     * assert(prefix_sum[2] == 5);
+     * assert(prefix_sum[3] == 10);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto alphabet_prefix_sum_sizes() noexcept
+    {
+        constexpr auto value_size = alphabet_sum_size_v<alphabets_t...>;
+        using rank_t = detail::min_viable_uint_t<value_size>;
+
+        constexpr size_t N = sizeof...(alphabets_t) + 1;
+        using array_t = std::array<rank_t, N>;
+
+        array_t prefix_sum{0, alphabet_size_v<alphabets_t>...};
+        for (auto i = 1u; i < N; ++i)
+            prefix_sum[i] = static_cast<rank_t>(prefix_sum[i] + prefix_sum[i-1]);
+
+        return prefix_sum;
+    }
+
+    /*!\brief Returns an fixed sized map at compile time where the key is the rank
+     * of alphabet_t and the value is the corresponding char of that rank.
+     * \sa value_to_char_table
+     *
+     * ```cpp
+     * constexpr auto table1 = value_to_char_table_I<5, char>(dna4{});
+     * assert(table1.size() == 5);
+     * assert(table1[0] == 'A');
+     * assert(table1[1] == 'C');
+     * assert(table1[2] == 'G');
+     * assert(table1[3] == 'T');
+     * assert(table1[4] == '\0');
+     * ```
+     */
+    template <size_t max_value_size, typename alphabet_t>
+    static constexpr auto value_to_char_table_I(alphabet_t alphabet) noexcept
```

Comment:
The capital `_I` is not compliant with our naming conventions. Should it be a Roman numeral and do we make an exception for this?